app/controllers/admin/reports_controller.rb

```diff
@@ -57,6 +63,21 @@ def filter_params
       params.slice(*ReportFilter::KEYS).permit(*ReportFilter::KEYS)
     end
 
+    def outdated_filter?
+      params.include?(:by_target_domain) || params.include?(:resolved)
+    end
+
+    def redirect_to_new_filter
+      by_target_domain = params.delete(:by_target_domain)
+      resolved = params.delete(:resolved)
+
+      redirect_to admin_reports_path filter_params.merge({
+        search_type: 'target',
+        search_term: by_target_domain,
+        status: resolved == '1' ? 'resolved' : nil,
+      })
+    end
```

Comment:
@mjankowski hey, is there a "better" way to do something like this?

The assumption is that moderators likely have links bookmarked to the previously valid filters that have been replaced by search and status filters, so we should keep those filters working by redirecting.

Reply:
I think having the request specs here (like you do) and then either something like this in the controller, or something in the routes (which detects the old style requests and redirects them from there) are both fine.

Maybe a TODO comment somewhere in here to remove the redirect after some reasonable amount of time or hard version bump or something.

Reply:
I'm not sure when / if we'd hard deprecate. Maybe a "# TODO: Delete for 5.0" or something?